### ISSUE TITLE
Unreliable update of image when setting Position in DotNet.Forms.VlcControl

### ISSUE THREAD

**Issue by perkalkyl** [ZeBobo5/Vlc.DotNet#533]
I have an issue / a question (pick one) about Vlc.DotNet.

# Generic information

- **Vlc.DotNet version** : 2.2.1
- **Vlc.DotNet project used** : WinForms
- **libvlc version** : x86) + NuGet
- **.net version** : 4.6.1
- **Project language** : C#)
- **Project build architecture** : AnyCPU (prefer 32bits))
- **Operating system** : (Windows 10) (x64)

# Summary
Hi!
I am trying to use the Forms.VlcControl in my application, and i need to be able to carefully scroll to find a certain frame or position. From a code point of view, all works fine. I can set the exact position, and vlc shows that exact position if i request the current position. If I play, the video starts from the set position. But while searching for the correct frame via updating set position, the video window is rarely updated to reflect the change. My need is to be able to search a position in pause mode to be able to find the exact frame.
Do i need to implement vlc some other way to achieve this? I would love to stick with the control you built!

- [ ] I confirm that my issue doesn't happen in VLC itself.

**Comment by jeremyVignelles** [comment]
What kind of stream? do you have the same behavior with vlc itself?
I'm afraid there's nothing I can do about it since it looks like a libvlc issue

**Comment by mfkl** [comment]
Can you share logs as well as repro code?

**Comment by perkalkyl** [comment]
Hi Jeremy, and thanx for fast response :-)
No, the problem does not occurr in VlcPlayer. The stream is an mxf file or mp4 file (usually) but it doesn't seem to matter what kind of stream it is. Well, maybe I should have stated that it is a video file :-)
If I open the file in VlcPlayer and pause the playback, i can move the scrollbar position very accurately and the image follows. If I try the same with my application, the video updates sometimes, and sometimes not at all. But as said, if i playback agin, the video will play from whatever point i selected with my scrollbar. It seems that the control does everything right, except not updating the actual panel it is displayed in while in pause. I have tried to add an .Refresh() on the vlc control with no success.

**Comment by mfkl** [comment]
What if you call `libvlc_media_player_next_frame`?

Or `Play` and then immediately `Pause`? Kind of a hack, but may be sufficient for your needs.

**Comment by perkalkyl** [comment]
Well, nextframe didn't do anything. The Play/Pause hack KINDA works, except it is very unreliable as to if it will pause or not. Sometimes it does, sometimes it keeps on playing

**Comment by jeremyVignelles** [comment]
I'd try to do something like
```
OnPlaying += PauseOnPlaying;
Play()

PauseOnPlaying(s,e){
 SetPause(true);
 OnPlaying-=PauseOnPlaying;
}
```

Still a hack though

**Comment by perkalkyl** [comment]
Thanx for your suggestion Jeremy
That idea actually made the response worse than the 
Play();
Pause();

idea. I guess it has to do with threading.. The play and pause commands, are they performed async or synced?
I am wondering if the only solution would be to do the whole buffering grabbing using the vlc control and not the Forms.VlcControl.. It's just that the Forms.VlcControl works so great in all other aspects

**Comment by jeremyVignelles** [comment]
How did you try the NextFrame solution? is there such a thing in Vlc.DotNet?

Play and Pause are sent synchronously to vlc. Then vlc posts the commands to its internal threads and this operation is asynchronous.

**Comment by perkalkyl** [comment]
Jeremy, I reached the NextFrame via the .DotNet.Forms.VlcControl.VlcMediaPlayer property

**Comment by mfkl** [comment]
What if you try to set the `Time` instead of the `Position`?

**Comment by jeremyVignelles** [comment]
@perkalkyl I don't have time to reproduce your issue right now. Would you mind publishing a sample project on github so that I can try it out?

**Comment by perkalkyl** [comment]
@mfkl It doesn't matter if I use Position or Time. They just seem to produce two different ways of describing the "cursor" position in the stream. Position is a completed float (0.0 - 1.0), and Time is a millisecond count (0  -  stream duration).

**Comment by perkalkyl** [comment]
@jeremyVignelles I am putting together a sample. But i need to examine my code before i burden you with this. The sample (that obviously is way lighter than my real app) behaves much better. Not perfect, but better, which might indicate that the problem lies in my code. I will get back with the result.
Also @mfkl, it seems that setting the position is much more secure than setting the time. If using the time, it goes into a deadlock very easily. It might have to do with the speed at which I update the position/time.

**Comment by jeremyVignelles** [comment]
Did you manage to make a sample?

**Comment by perkalkyl** [comment]
Hi @jeremyVignelles 
I did a sample code, and in that code the playback head performs as good as can be expected. I have concluded that setting the Time is way more unreliable than setting the position. But that's fine. So as it is obvious that the issue lies in my application handling of the control (it's kinda complex), I didn't want to burden you chasing ghosts. If I manage to lock down the exact reason for the slow or nonexisting update, I will make an update to this thread, or if it is closed, make a new one.

BR
Per
